core/templates/dev/head/pages/question_editor/QuestionEditorDirective.js

```diff
@@ -97,51 +96,83 @@ oppia.directive('questionEditor', [
             $scope.stateEditorInitialized = true;
           };
 
+          var _updateQuestion = function(updateFunction) {
+            var oldQuestionStateData =
+              angular.copy($scope.question.getStateData());
+            updateFunction();
+            QuestionUpdateService.setQuestionStateData(
```

Comment:
Yeah, I already tried that, it didn't work. I can file an issue?

Reply:
I think it's fine, since the question is bound to scope, we are effectively updating the scope values and the actual question object separately, right?
But since, the questionData field of question is also bound to scope, won't doing `updateFunction()` change the question object also? So effectively the `setQuestionStateData()` function only adds a change object, right? Since question was already updated.

Also, yeah, I don't think we can do it similar to other editors as question editor does not have a separate page for itself, nor a state service, so maybe this is fine? What do you think?

Reply:
Just to give a third-party perspective on this: I'm kind of confused, reading the code. It looks like stateData is being modified in updateFunction() and then it's being modified again here (from the name setQuestionStateData)?

Should setQuestionStateData be part of updateFunction instead? If you need to reference oldQuestionStateData you can probably do currying (with bind() or something). But as it stands currently, I'm kinda confused about what's going on here.

(Happy to chat if useful.)

Reply:
Okay, I updated it so that there's not as much redundancy, take a look?